UnityProject/Assets/Scripts/ScriptableObjects/TimedGameEvents/TimedGameEventSO.cs

```diff
@@ -16,11 +13,26 @@ public class TimedGameEventSO : ScriptableObject
 		public string EventName;
 		[TextArea(10, 20)]
 		public string EventDesc;
-		
+
 		public Sprite EventIcon;
 
 		//Delete the object when the timed event is not happening
 		public bool deleteWhenNotTime = true;
+
+		public virtual IEnumerator EventStart()
+		{
+			yield break;
+		}
+		public virtual IEnumerator OnRoundEnd()
+		{
+			yield break;
+		}
+
+		public virtual void Clean()
+		{
+			//Implement your own custom logic to clean up your scriptable objects here.
+			//This will be called when the round ends but you can call it whenever needed.
+		}
```

Comment:
Could these be abstract instead? So they aren't forgotten.

Reply:
good point

Reply:
![image](https://user-images.githubusercontent.com/34368774/193474598-c13581b9-b6d0-4127-ab88-905ef3e1bbd8.png)

You can't have an abstract method in a scriptable object

Reply:
if you say "just make the class abstract"
![image](https://user-images.githubusercontent.com/34368774/193474667-c0e3135f-e33c-41e3-9766-632b784e058b.png)

it breaks events

Reply:
also, some events don't need these methods

Reply:
Unity moment.

Reply:
> You can't have an abstract method in a scriptable object

not entirely true. I always do an abstract base class extending scriptable objects and then my concrete classes extending that one. Your Events break when you make it abstract because by creating assets from the abstract, you are trying to instantiate it. 

TLDR: make abstract parent and extend it with your concrete classes that will generate .asset files.